bls.go

```diff
@@ -13,22 +13,15 @@ import "C"
 
 // Hash computes the digest of a message
 func Hash(message Message) Digest {
-	// prep request
-	cMessage := C.CBytes(message)
-	defer C.free(cMessage)
-	cMessagePtr := (*C.uchar)(cMessage)
-	cMessageLen := C.size_t(len(message))
-
-	// call method
-	resPtr := (*C.HashResponse)(unsafe.Pointer(C.hash(cMessagePtr, cMessageLen)))
-	defer C.destroy_hash_response(resPtr)
-
-	// prep response
-	var digest Digest
-	digestSlice := C.GoBytes(unsafe.Pointer(&resPtr.digest), DigestBytes) // nolint: staticcheck
-	copy(digest[:], digestSlice)
-
-	return digest
+	resp := generated.FilHash(string(message), uint(len(message)))
+	resp.Deref()
```

Comment:
what are the rules around these `Deref` calls?

Reply:
If Go receives a pointer from C, the caller must call `Deref()` on that pointer before its fields will be populated with a shallow copy of values from the C heap. `Deref()` will only unpack to a slice field if that slice has been initialized first, so sometimes two `Deref()` calls will be necessary. The `proofs.go` file should include sufficient examples such that further work in this area will be straightforward.